installers/rpm-code/codes/mantid.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+mantid_main() {
+    codes_yum_dependencies \
+        eigen3-devel \
+        OCE-devel \
+        jemalloc-devel \
+        jsoncpp-devel \
+        muParser-devel \
+        poco-devel
+    codes_dependencies common boost ipykernel
+    install_pip_install pre-commit pylint sphinx_bootstrap_theme
+    mantid_nexus_install
+    # Mantid installs a great deal and requires its own specific environment variables to run right
+    # (ex. see the mantidpython script it creates). So, install it in its own subdirectory so it
+    # doesn't interfere with other libs.
+    local p="${codes_dir[lib]}/mantid"
+    mantid_install "$p"
+    mantid_ipykernel "$p" "$(mantid_install_rsmantid "$p")"
+}
+
+mantid_install() {
+    local install_prefix=$1
+    codes_download mantidproject/mantid
+    mantid_patch_eigen_cmake
+    codes_cmake \
+        -DBOOST_ROOT="${codes_dir[prefix]}" \
+        -DCMAKE_INSTALL_PREFIX="$install_prefix" \
+        -DENABLE_WORKBENCH=OFF \
+        -DMANTID_FRAMEWORK_LIB=BUILD \
+        -DMANTID_QT_LIB=OFF \
+        -DNEXUS_CPP_LIBRARIES="${codes_dir[lib]}"/libNeXusCPP.so \
+        -DNEXUS_C_LIBRARIES="${codes_dir[lib]}"/libNeXus.so \
+        -DNEXUS_INCLUDE_DIR="${codes_dir[include]}"/nexus/ \
+        '-GUnix Makefiles'
+    codes_cmake_build
+    codes_make_install
+}
+
+mantid_install_rsmantid() {
+    local install_d=$1
+    local s=rsmantid
+    codes_download_module_file "$s.sh"
+    local l=$install_d/bin/mantidpython
+    install -m 555 /dev/stdin "${codes_dir[bin]}/$s" <<EOF
+#!/bin/bash
+set -euo pipefail
+exec '$l' --classic "\$@"
+EOF
+    echo "$l"
+}
+
+mantid_ipykernel() {
+    local INSTALLDIR=$1
+    local mantidpython=$2
+    local LOCAL_PRELOAD LOCAL_PYTHONPATH LOCAL_LDPATH
+    eval "$(grep -P '^LOCAL_\w+=' "$mantidpython")"
+    python -m ipykernel install \
+        --display-name 'Mantid Python' \
+        --name pymantid \
+        --user \
+        --env PYENV_VERSION py3 \
```

Comment:
I don't think PYENV_VERSION is required, because it's the default. We can avoid distributing the coupling with `py3` by not specifying it here.

Reply:
39360a2